@xen-orchestra/lite/src/components/ui/UiStatusPill.vue

```diff
@@ -0,0 +1,43 @@
+<template>
+  <UiIcon :icon class="icon" :class="props.state" />
+</template>
+
+<script lang="ts" setup>
+import UiIcon from '@/components/ui/icon/UiIcon.vue'
+import { computed } from 'vue'
+import { faCheckCircle, faCircleMinus, faCircleXmark } from '@fortawesome/free-solid-svg-icons'
+import type { Color } from '@/types'
+
+const props = defineProps<{
+  state: Color
+}>()
+
+const icon = computed(() => {
+  switch (props.state) {
+    case 'error':
+      return faCircleXmark
+    case 'warning':
+      return faCircleMinus
+    case 'success':
+      return faCheckCircle
+    default:
+      return faCircleXmark
+  }
+})
```

Comment:
Instead of a `switch`, you can use a configuration object to map `icon` and `color` to `state`.
```suggestion
const states = {
  success: { icon: faCheckCircle, color: 'success' },
  partial: { icon: faCircleMinus, color: 'warning' },
  failure: { icon: faCircleXmark, color: 'error' },
}

const icon = computed(() => states[props.state].icon)
const color = computed(() => states[props.state].color as Color)
```